servant-server/src/Servant/Server/Internal.hs

```diff
@@ -134,8 +135,8 @@ processMethodRouter :: forall a. ConvertibleStrings a B.ByteString
                     -> Maybe [(HeaderName, B.ByteString)]
                     -> Request -> RouteResult Response
 processMethodRouter handleA status method headers request = case handleA of
-  Nothing -> failWith UnsupportedMediaType
-  Just (contentT, body) -> succeedWith $ responseLBS status hdrs bdy
+  Nothing -> FailFatal err406
```

Comment:
If you rule it out, though, it still means I cannot do
```
type MyAPI c = ... Put '[c] T ...
```
and then define severs for `MyAPI Foo` and `MyAPI Bar` separately and subsequently `:<|>` them.

So you'd argue that because we cannot rule out that types might be different, we should rule out the whole pattern, and because we cannot rule that out either, we effectively make all but the first branch unreachable?